cmd/ztest/ztest.c

```diff
@@ -6839,6 +6839,23 @@ ztest_run(ztest_shared_t *zs)
 	}
 	zs->zs_enospc_count = 0;
 
+	/*
+	 * If we were in the middle of ztest_device_removal() and were killed
+	 * we need to ensure the removal and scrub complete before running
+	 * any tests that check ztest_device_removal_active. The removal will
+	 * be restarted automatically when the spa is opened, but we need to
+	 * initate the scrub manually if it not already in progress.
```

Comment:
This does the scrub even if we were not in the middle of ztest_device_removal(); it scrubs if we completed a removal (and completed the scrub).  I think this is fine but can you update the comment to acknowledge that we are doing a scrub if there is a previously completed removal, in which case ztest_device_removal() may have already done a scrub, but we don't know for sure so we do it again here.